lib/Extractor/KLEEBuilder.cpp

```diff
@@ -367,6 +367,46 @@ class KLEEBuilder : public ExprBuilder {
       unsigned Index = Ops[1]->Val.getZExtValue();
       return get(Ops[0]->Ops[Index]);
     }
+    case Inst::SAddSat: {
+      ref<Expr> add = AddExpr::create(get(Ops[0]), get(Ops[1]));
+      auto sextL = SExtExpr::create(get(Ops[0]), I->Width + 1);
+      auto sextR = SExtExpr::create(get(Ops[1]), I->Width + 1);
+      auto addExt = AddExpr::create(sextL, sextR);
+      auto smin = klee::ConstantExpr::alloc(llvm::APInt::getSignedMinValue(I->Width));
+      auto smax = klee::ConstantExpr::alloc(llvm::APInt::getSignedMaxValue(I->Width));
+      auto sminExt = SExtExpr::create(smin, I->Width + 1);
+      auto smaxExt = SExtExpr::create(smax, I->Width + 1);
+      auto pred = SleExpr::create(addExt, sminExt);
+
```

Comment:
I feel like this is a pretty complicated way to detect signed overflow. why not reuse our signed overflow code or else just look for the result having a different sign than the inputs (you don't need to do any sign extension in that case)

Reply:
this (and others) is taken verbatim from alive2 to keep it consistent with it; I didn't want to come up with any clever way for this (with potential risks of getting wrong). Moreover, we want to get rid of KLEE anyways according to you; so, i didn't want to invest much time on it.

Reply:
ya, ok ok :)